.github/workflows/custom-actions/set-commit-status/action.yaml

```diff
@@ -0,0 +1,39 @@
+name: Set Commit Status
+
+description: |
+  A reusable action that sets the commit status.  This is used to set PR status
+  from workflows with non-PR triggers (such as manually-triggered workflows).
+
+inputs:
+  context:
+    description: An arbitrary string to identify the status check.
+    required: true
+  state:
+    description: Either "pending", "error", "success", or "failure".
+    required: true
+  ref:
+    description: A git ref for which to set the commit status.  For PRs, use the head commit, not the merge commit.  Defaults to HEAD.
+    required: false
+    default: HEAD
+  token:
+    description: A GitHub access token.
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Report Commit Status
+      shell: bash
+      run: |
+        # This is the URL to view this workflow run on GitHub.  It will be
+        # attached to the commit status, so that when someone clicks "details"
+        # next to the status on the PR, it will link to this run where they can
+        # see the logs.
+        RUN_URL="$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/actions/runs/$GITHUB_RUN_ID"
+        SHA1=$(git rev-parse "${{ inputs.ref }}")
+        GITHUB_TOKEN=${{ inputs.token }} gh api \
+          -X POST \
+          -F "context=${{ inputs.context }}" \
+          -F "state=${{ inputs.state }}" \
+          -F "target_url=$RUN_URL" \
+          "repos/${{ github.repository }}/statuses/$SHA1"
```

Comment:
You are correct.  The only reason is that I cobbled this together over several iterations, drawing information from several sources, and I ended up with mixed styles because of it.

I've switched to using context variables where possible.